contrib/devtools/test-symbol-check.py

```diff
@@ -5,47 +5,58 @@
 '''
 Test script for symbol-check.py
 '''
+import os
 import subprocess
+from typing import List
 import unittest
 
-def call_symbol_check(cc, source, executable, options):
-    subprocess.run([cc,source,'-o',executable] + options, check=True)
+from utils import determine_wellknown_cmd
+
+def call_symbol_check(cc: List[str], source, executable, options):
+    # This should behave the same as AC_TRY_LINK, so arrange well-known flags
+    # in the same order as autoconf would.
+    #
+    # See the definitions for ac_link in autoconf's lib/autoconf/c.m4 file for
+    # reference.
+    env_flags: List[str] = []
+    for var in ['CFLAGS', 'CPPFLAGS', 'LDFLAGS']:
+        env_flags += filter(None, os.environ.get(var, '').split(' '))
+
+    subprocess.run([*cc,source,'-o',executable] + env_flags + options, check=True)
     p = subprocess.run(['./contrib/devtools/symbol-check.py',executable], stdout=subprocess.PIPE, universal_newlines=True)
+    os.remove(source)
+    os.remove(executable)
     return (p.returncode, p.stdout.rstrip())
 
-def get_machine(cc):
```

Comment:
this line is re-landed back with bitcoin/bitcoin#22381 but seems as this changes are missing

Reply:
Good catch! The cause behind was out-of-order backports which removed invocations, did cleanups, then reintroduced it, the backports swapped the order of the last two making it remove the function but keeping the invocation (the reintroduction didn't stick through as the messed up order meant that it was already present when it should've been absent)